src/math/LinearCombinationKernel.cc

```diff
@@ -107,6 +107,24 @@ std::shared_ptr<Kernel> LinearCombinationKernel::clone() const {
     return retPtr;
 }
 
+std::shared_ptr<Kernel> LinearCombinationKernel::resized(int width, int height) const {
+    KernelList kernelList;
+
+    for (KernelList::const_iterator kIter = _kernelList.begin(), kEnd = _kernelList.end(); kIter != kEnd;
+         ++kIter) {
+        std::shared_ptr<Kernel> basisKernelPtr = (*kIter)->resized(width, height);
+        kernelList.push_back(basisKernelPtr);
```

Comment:
I'm surprised my personal opinion on `auto` hasn't come up in conversation before: it is easier to write, but harder to read.  Consider my typing out the type as a gift, so you don't need to go read the declaration of getKernelList().